### ISSUE TITLE
Unicorn doesn't support x86 MOVDIRI instruction?

### ISSUE THREAD

**Issue by smmalis37** [unicorn-engine/unicorn#1777]
Attempting to run the instruction bytes `0f 38 f9 00` returns an INSN_INVALID error. I'd expect it to just run the instruction.

**Comment by wtdcode** [comment]
Please check your CPUID to see if it's supported. Note you could always change CPU model.

**Comment by smmalis37** [comment]
It is supported on my CPU. How do I change CPU model in the Rust bindings?

**Comment by wtdcode** [comment]
> It is supported on my CPU. How do I change CPU model in the Rust bindings?

I mean, you should check if the instruction is supported or not for the emulated CPU.

**Comment by smmalis37** [comment]
How do I control that? I don't see anything related to the emulated CPU model, I just call Unicorn::new() and write things into memory and set up my hooks.

**Comment by smmalis37** [comment]
Did a bit more digging, are you referring to uc_ctl_set_cpu_model? It is not currently included in the Rust bindings, nor is the list of supported cpu models. In fact, I only see it called in the Go and Java bindings, and none of the others.

**Comment by wtdcode** [comment]
> Did a bit more digging, are you referring to uc_ctl_set_cpu_model? It is not currently included in the Rust bindings, nor is the list of supported cpu models. In fact, I only see uc_ctl_set_cpu_model called in the Go and Java bindings, and none of the others.

Yes correct because Rust bindings is more of maintained by the community and I lack time to fix it...
